src/services/review/RepoCheckerService.ts

```diff
@@ -53,17 +55,69 @@ export default class RepoCheckerService {
   constructor({
     siteMemberRepository,
     repoRepository,
-    sitesService,
     gitFileSystemService,
     git,
   }: RepoCheckerServiceProps) {
     this.siteMemberRepository = siteMemberRepository
-    this.sitesService = sitesService
+
     this.gitFileSystemService = gitFileSystemService
     this.git = git
     this.repoRepository = repoRepository
   }
 
+  isCurrentlyLocked(repo: string): ResultAsync<true, SiteCheckerError> {
```

Comment:
We seem to be having multiple locks. There is a DDB, then we introduced a GIT efs lock on repo level and this is a new lock. Is there a way to simplify this?

Reply:
you bring a good point, but the three locks here solve varying levels of locking.
1. git locks is for not each git operation
2. ddb lock is at a repo level (eg it is possible for a save to end up committing twice)
3. repo checking lock, which is at this time, not related to any repo changes. While that could be linked in the furture, i rather keep it simple for to get the initial traction

Reply:
okay fair enough